js/sdk/src/utils/downloadManager.js

```diff
@@ -0,0 +1,169 @@
+import ID3Writer from 'browser-id3-writer'
+import { saveAs } from 'file-saver'
+import axios from 'axios'
+import { logEvent } from './event'
+
+export const downloadAs = async (
+  url,
+  releasePubkey,
+  artwork,
+  artist,
+  title,
+  description,
+  link,
+  setDownloadId,
+  enqueueSnackbar,
+  walletAddress,
+  hubPubkey
+) => {
+  setDownloadId(releasePubkey)
+  enqueueSnackbar('Downloading Release', { variant: 'info' })
+
+  logEvent('track_download_dashboard', 'engagement', {
+    publicKey: releasePubkey,
+    hub: hubPubkey,
+    wallet: walletAddress,
+  })
+  try {
+    const download = await axios
+      .get(url, {
+        method: 'GET',
+        mode: 'cors',
+        headers: {
+          'Content-Type': 'application/octet-stream',
+        },
+        responseType: 'blob',
+      })
+      .then(async (res) => {
+        const buffer = await res.data.arrayBuffer()
+        let image = artwork
+        if (image) {
+          image = await fetch(image).then((r) => r.blob())
+          image = await new Response(image).arrayBuffer()
+        }
+        const writer = new ID3Writer(buffer)
+        writer.setFrame('TIT2', title)
+        writer.setFrame('TPE1', [artist])
+        writer.setFrame('WPAY', link)
+        writer.setFrame('COMM', {
+          description: description,
+          text: `Downloaded from Nina Protocol: ${link}`,
+          language: 'eng',
+        })
+        writer.setFrame('APIC', {
+          type: 3,
+          data: image,
+          description: 'Cover',
+        })
+        writer.addTag()
+        const blob = writer.getBlob()
+        saveAs(blob, `${title}.mp3`)
```

Comment:
![Screenshot 2023-03-21 at 2 25 25 PM](https://user-images.githubusercontent.com/2960410/226706168-091e92f2-920f-48a8-958d-10d839ed5792.png)
it appears that filenames with "/" cause folders to be made - you should sanitize filenames before saving them